scripts/us_bls/jolts/bls_jolts.py

```diff
@@ -0,0 +1,346 @@
+name: "InvoluntarySeparation"
+description: "Describes a worker that involuntarily separated from a company"
+typeOf: dcid:JobChangeEventEnum
+specializationOf: dcs:Separated
+
+Node: dcid:OtherSeparation
+name: "OtherSeparation"
+description: "Describes a worker that separated from a company for other reasons"
+typeOf: dcid:JobChangeEventEnum
+specializationOf: dcs:Separated
+  """
+
+  template_bls = """
+Node: dcid:NAICS/JOLTS_{JOLTS_CODE}
+typeOf: NAICSEnum
+name: {JOLTS_NAME}
+  """
+
+  with open(STATISTICAL_VARIABLE_FILE, "w+", newline="") as f_out:
+    f_out.write(required_new_nodes)
+
+    for _, new_code in CODE_MAPPINGS.items():
+      if ":" in new_code:
+        jolts_code, jolts_name = new_code.split(":")
+        f_out.write(template_bls.format_map({"JOLTS_CODE": jolts_code, "JOLTS_NAME": jolts_name}))
+
+def create_statistical_variables(jolts_df, schema_mapping):
+  """ Creates Statistical Variable nodes.
+
+    A new statistical industry is needed for each of the 6 job variables and for every industry
+    The industry codes may be either NAICS or BLS_JOLTS aggregations. The schema_mapping is used 
+    for additional information for each of the 6 job variables. These new variables are written
+    to the statistical variables mcf file.
+
+    Args:
+      jolts_df: The dataframe of BLS Jolts data created by generate_cleaned_dataframe
+      schema_mapping: The schema mapping created by generate_cleaned_dataframe
+  """
+
+  template_stat_var = """
+  Node: dcid:{STAT_CLASS}_NAICS_{INDUSTRY}
+  typeOf: StatisticalVariable
+  populationType: {POPULATION}
+  jobChangeEvent: dcs:{JOB_CHANGE_EVENT}
+  statType: dcs:measuredValue
+  measuredProperty: dcs:count
+  naics: dcid:NAICS/{INDUSTRY}
+  """
+
+  # Map industry and seasonal adjustment to statistical variable name
+  adjustment_types = [("Adjusted", "BLSSeasonallyAdjusted"), ("Unadjusted", "BLSSeasonallyUnadjusted")]
+
+  # Output the schema mapping to a new file
+  with open(STATISTICAL_VARIABLE_FILE, "a+", newline="") as f_out:
+    for schema_name, pop_type, job_change_event, _ in schema_mapping:
+
+        unique_industries = list(jolts_df['industry_code'].unique())
+
+        for industry_code in unique_industries:
+          for adjusted_dcid_map, adjusted_schema in adjustment_types:
+            # Create new schema object
+            stat_var_schema = template_stat_var
+
+            # Remove separation type entry if not includes
+            if job_change_event == "":
+              stat_var_schema = stat_var_schema.replace("jobChangeEvent: dcs:{JOB_CHANGE_EVENT}\n", "")
+
+            # Replace all other fields
+            stat_var_schema = stat_var_schema.replace("{STAT_CLASS}", schema_name)   \
+                                            .replace("{INDUSTRY}", industry_code)   \
+                                            .replace("{ADJUSTMENT}", adjusted_dcid_map)   \
+                                            .replace("{BLS_ADJUSTMENT}", adjusted_schema)   \
```

Comment:
Looks like these two lines leftover (unused).

Reply:
Or maybe you were about to use them and then didn't fully go through... the measurementQualifier IS supposed to be in the StatVar (not the StatVarObs).

Reply:
Now needed for qualifier.